src/Microsoft.AspNet.Mvc.Core/Filters/AuthorizationFilter.cs

```diff
@@ -53,11 +32,11 @@ public string Roles
             var authService = httpContext.RequestServices.GetRequiredService<IAuthorizationService>();
 
             // Build a policy for the requested roles if specified
-            if (_rolesSplit != null)
+            if (Roles != null && Roles.Any())
             {
                 var rolesPolicy = new AuthorizationPolicyBuilder();
-                rolesPolicy.RequiresRole(_rolesSplit);
-                if (!await authService.AuthorizeAsync(rolesPolicy.Build(), httpContext, context))
+                rolesPolicy.RequiresRole(Roles.ToArray());
```

Comment:
Fair enough, I'll add an overload to Policy builder which also allows `IEnumerable<string>`